File: backend/models/Employee.js
const mongoose = require('mongoose');

const employeeSchema = new mongoose.Schema({
  name: { type: String, required: true },
  phone: { type: String, required: true },
  department: { type: String, required: true },
  employeeCode: { type: String, unique: true, sparse: true }
}, { timestamps: true });

module.exports = mongoose.model('Employee', employeeSchema);

File: mobile/lib/models/employee.dart
class Employee {
  final String id;
  final String name;
  final String department;
  final String employeeCode;

  Employee({
    required this.id,
    required this.name,
    required this.department,
    required this.employeeCode,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'],
      name: json['name'],
      department: json['department'],
      employeeCode: json['employeeCode'],
    );
  }
}